test/parallel/test-util-inspect.js

```diff
@@ -2551,6 +2547,293 @@ assert.strictEqual(
+    '          stack: [Getter/Setter: <Inspection threw (Error: Oops\n' +
+    '              at get foo (/foo/node_modules/foo.js:2:7))>],\n' +
+    '          message: [Getter: <Inspection threw (Error: Oops\n' +
+    '              at get foo (/foo/node_modules/foo.js:2:7))>]\n' +
+    '        })>]\n' +
+    '      })>]\n' +
+    '    })>]\n' +
+    '  })>]\n' +
+    '}'
+  );
+}
+
+// Property getter throwing an error whose own getters throw that same error (infinite recursion).
+{
+  const badError = new Error();
+
+  const throwingGetter = {
+    __proto__: null,
+    get() {
+      throw badError;
+    },
+    configurable: true,
+    enumerable: true,
+  };
+
+  Object.defineProperties(badError, {
+    name: throwingGetter,
+    message: throwingGetter,
+    stack: throwingGetter,
+    cause: throwingGetter,
+  });
+
+  const thrower = {
+    get foo() { throw badError; }
+  };
+
+  assert.strictEqual(
+    inspect(thrower, { getters: true, depth: Infinity }),
+    '{\n' +
+    '  foo: [Getter: <Inspection threw (<ref *1> [object Error] {\n' +
+    '    stack: [Getter/Setter: <Inspection threw ([Circular *1])>],\n' +
+    '    name: [Getter: <Inspection threw ([Circular *1])>],\n' +
+    '    message: [Getter: <Inspection threw ([Circular *1])>],\n' +
+    '    cause: [Getter: <Inspection threw ([Circular *1])>]\n' +
+    '  })>]\n' +
+    '}'
+  );
+}
+
+// Property getter throwing uncommon values.
+{
+  assert.strictEqual(
+    inspect({
+      // eslint-disable-next-line no-throw-literal
+      get foo() { throw undefined; }
+    }, { getters: true }),
+    '{ foo: [Getter: <Inspection threw (undefined)>] }'
+  );
+  assert.strictEqual(
+    inspect({
+      // eslint-disable-next-line no-throw-literal
+      get foo() { throw null; }
+    }, { getters: true }),
+    '{ foo: [Getter: <Inspection threw (null)>] }'
+  );
+  assert.strictEqual(
+    inspect({
+      // eslint-disable-next-line no-throw-literal
+      get foo() { throw 'string'; }
+    }, { getters: true }),
+    "{ foo: [Getter: <Inspection threw ('string')>] }"
+  );
+  assert.strictEqual(
+    inspect({
+      // eslint-disable-next-line no-throw-literal
+      get foo() { throw true; }
+    }, { getters: true }),
+    '{ foo: [Getter: <Inspection threw (true)>] }'
+  );
+  assert.strictEqual(
+    inspect({
+      // eslint-disable-next-line no-throw-literal
+      get foo() { throw {}; }
+    }, { getters: true }),
+    '{ foo: [Getter: <Inspection threw ({})>] }'
+  );
+  assert.strictEqual(
+    inspect({
+      // eslint-disable-next-line no-throw-literal
+      get foo() { throw { get message() { return 'Oops'; } }; }
+    }, { getters: true }),
+    "{ foo: [Getter: <Inspection threw ({ message: [Getter: 'Oops'] })>] }"
+  );
+  assert.strictEqual(
+    inspect({
+      get foo() { throw Error; }
+    }, { getters: true }),
+    '{ foo: [Getter: <Inspection threw ([Function: Error])>] }'
+  );
+}
```

Comment:
Let's we make this more readable by using a array

```suggestion
[
  { val: undefined, expected: '{ foo: [Getter: <Inspection threw (undefined)>] }' },
  { val: null, expected: '{ foo: [Getter: <Inspection threw (null)>] }' },
  { val: true, expected: '{ foo: [Getter: <Inspection threw (true)>] }' },
  { val: 1, expected: '{ foo: [Getter: <Inspection threw (1)>] }' },
  { val: 1n, expected: '{ foo: [Getter: <Inspection threw (1n)>] }' },
  { val: Symbol(), expected: '{ foo: [Getter: <Inspection threw (Symbol())>] }' },
  { val() {}, expected: '{ foo: [Getter: <Inspection threw (Function val())>] }' },
  { val: 'string', expected: "{ foo: [Getter: <Inspection threw ('string')>] }" },
  { val: [], expected: "{ foo: [Getter: <Inspection threw ({})>] }" },
  { val: { get message() { return 'Oops'; } }, expected: "{ foo: [Getter: <Inspection threw ({ message: [Getter: 'Oops'] })>] }" },
  { val: Error, expected: "{ foo: [Getter: <Inspection threw ([Function: Error])>] }" },
].forEach(({val, expected}) => {
    assert.strictEqual(
      inspect({
        get foo() { throw val; }
      }, { getters: true }),
      expected,
    );
});

```